src/sklearn_evaluation/plot/calibration.py

```diff
@@ -29,16 +29,369 @@
+    if any((val is None for val in (mean_predicted_value, fraction_of_positives))):
+        raise ValueError(
+            "Mean_predicted_value and fraction_of_positives"
+            " are "
+            "needed to plot Calibration Curve"
+        )
+
+    # validate input metrics of same length
+    if len(mean_predicted_value) != len(fraction_of_positives):
+        raise ValueError(
+            "mean_predicted_value and fraction_of_positives lengths should correspond. "
+            f"Received: mean_predicted_value {len(mean_predicted_value)} "
+            f"!= fraction_of_positives {len(fraction_of_positives)}"
+        )
+
+
+@modify_exceptions
+def _validate_raw_data(probabilities, y_true, clf_names):
+    if any((val is None for val in (probabilities, y_true))):
+        raise ValueError(
+            "probabilities and y_true" " are " "needed to plot Calibration Curve"
+        )
+
+    if not isinstance(probabilities, list):
+        raise ValueError("`probabilities` should be a list.")
+
+    if isiterofiter(y_true) and len(y_true) != len(probabilities):
+        raise ValueError(
+            f"y_true (Received : {len(y_true)}) and probabilities "
+            f"(Received : {len(probabilities)}) should have the "
+            "same size when y_true is an iterator of array-like objects"
+        )
+    if len(clf_names) != len(probabilities):
+        raise ValueError(
+            "Length {} of `clf_names` does not match length {} of"
+            " `probabilities`".format(len(clf_names), len(probabilities))
+        )
+
+
+def _plot_from_metrics(mpv, fop, clf_names, color, ax):
+    for mpv, fp, name, c in zip(mpv, fop, clf_names, color):
+        ax.plot(
+            mpv,
+            fp,
+            "s-",
+            label=name,
+            color=c,
+        )
+
+    return ax
+
+
+def _generate_colors(cmap, n_color):
+    return [plt.cm.get_cmap(cmap)(float(i) / n_color) for i in range(n_color)]
+
+
+class CalibrationCurve(AbstractPlot):
+
+    """
+    Parameters
+    ----------
+    mean_predicted_value : ndarray of shape (n_bins,) or smaller
+        The mean predicted probability in each bin.
+
+    fraction_of_positives : ndarray of shape (n_bins,) or smaller
+        The proportion of samples whose class is the positive class, in each
+        bin.
+
+    clf_names : list of str, optional)
+            A list of strings, where each string refers to the name of the
+            classifier that produced the corresponding probability estimates in
+            `probabilities`. If ``None``, the names "Classifier 1", "Classifier 2",
+            etc. will be used.
+
+    cmap : string or :class:`matplotlib.colors.Colormap` instance, optional
+            Colormap used for plotting the projection. View Matplotlib Colormap
+            documentation for available options.
+            https://matplotlib.org/users/colormaps.html
+
+    name : string, optional
```

Comment:
let's rename this to be consistent

```suggestion
    label : string, optional
```

Reply:
name is name of the plot. `label` in other plots refer to the curve labels (which in this case is `clf_names`).
Should I still rename?
@edublancas

Reply:
@edublancas ^^

Reply:
I see. so let's keep the same signature `calibration_curve` to have backward compatibility. For the OO interface let's delete `name` (we can think of a way to allow users to customize the plot's title that's standard across different plots) and rename `clf_names` to `label`